### ISSUE TITLE
Python API named parameters not always working

### ISSUE THREAD

**Issue by zx80** [duckdb/duckdb#7602]
### What happens?

It seems that SQL keywords are not accepted as named parameters.

### To Reproduce

```python
import duckdb
conn = duckdb.connect(":memory:")
conn.execute("SELECT $val", {"val": 42}).fetchone()
# (42,)
conn.execute("SELECT $value", {"value": 42}).fetchone()
# ParserException: Parser Error: syntax error at or near "value"
```

### OS:

Linux

### DuckDB Version:

0.8.1-dev23

### DuckDB Client:

Python

### Full Name:

Fabien Coelho

### Affiliation:

Mines Paris

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by zx80** [comment]
And others, e.g. `content` which is not an SQL keyword AFAICR.
Failing include: `left`, `select`, `table`, `join`…

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #7604
